docs/framework/30scoring/implementation.md

```diff
@@ -1,51 +1,187 @@
 ## Implementation Risk Scoring
+Implementation Risk Scoring is based on the [Implementation Risk](../20categories/30implementation/protocol-implementation-risk.md#protocol-implementation-risk) section.
 
-The following aspects of [Implementation Risk](../20categories/30implementation/protocol-implementation-risk.md#protocol-implementation-risk) are currently not taken into account when assessing the Implementation Risk Score: 
-[Mixing of Control and Data Plane](../20categories/30implementation/mixing-control-data-flow.md), 
-[Role Based Access Control](../20categories/30implementation/access-control.md),
-[Upgradability](../20categories/30implementation/upgrade.md),
-[Secret Storage](../20categories/30implementation/secret-storage.md),
-[Product Development Maturity](../20categories/30implementation/maturity.md),
-[Well Known Platform](../20categories/30implementation/known-platform.md),
-[Well Known Smart Contract Programming Language](../20categories/30implementation/known-language.md),
-[Formal Verification](../20categories/30implementation/formal-verification.md),
-[Testing](../20categories/30implementation/testing.md),
-[Open Source Code](../20categories/30implementation/open-source.md),
-[Verified Code on Block Explorer](../20categories/30implementation/verified-code.md),
-[Documentation](../20categories/30implementation/documentation.md),
-[Bug Bounty](../20categories/30implementation/bug-bounty.md),
-[Ability to Pause Project](../20categories/30implementation/pause.md),
-and 
-[Ability to Ban Addresses](../20categories/30implementation/ban-address.md).
+The equation for the *Implementation Risk Score* is:
 
+```
+Implementation Risk Score = Maximum of (Reducing Implementation Risk Score, Uncovering Extant Risk Score, and Responding to Materialized Risk Score)
+```
 
-The equation for the *Implementation Risk Score* is:
+### Reducing Implementation Risk
+```
+Reducing Implementation Risk Score = 
+ Mixing of Control and Data Plane Risk Score +
+ Role Based Access Control Risk Score +
+ Upgradable Risk Score +
+ Secret Storage Risk Score +
+ Product Development Maturity Risk Score +
+ Well Known Platform Risk Score +
+ Well Known Smart Contract Programming Language Risk Score
```

Comment:
I would suggest we remove these two from the risk scoring. Scoring all crosschain protocols that span non-EVM chains (Solana, Cosmos etc.) as inherently risky feels like an over-generalisation, and probably not very useful given most protocols support such chains.

Reply:
It would not be the whole protocol, just the bridge links that involve a risky platform or a risky programming language. What this reflects is the inherent increased risk of going between a not well known platform and a known platform; or using a programming language other than Solidity (with no non-library assembler)

Reply:
I have added clarifying text to the Well Known Smart Contract Programming Language Risk Score section, explaining that it depends on the source and target chain, and does not pertain to an entire system.

Reply:
So the reason we think less known platforms and smart contract programming languages could potentially be a source of risk is two-fold: 1) the language or platforms themselves might not be mature enough and might have vulnerabilities 2) developers and auditors might not be as familiar with them, and the applications that they build for these environments might have vulnerabilities. 

I think that while calling these out and discussing them in the implementation risk section can be informative, trying to concretely quantify them in a risk score, might not, for a few reasons:

1. Arguing that bridging to Cosmos or Solana is quantifiably riskier than bridging to any non-EVM chain, or that contracts written in Rust are likely to have more vulnerabilities than any written in Solidity feels imprecise and likely tenuous. With the exception of very nascent environments both of these are very difficult to confidently assert broadly, and make this measurement quite subjective. 

2. Our bias and experience might be more toward EVM-based chains and this might discount the wealth of developer tools, frameworks and resources in other ecosystems we might not be intimately familiar with.

3. The weight we give to this risk factor feels very high (I701 alone is 20). If I am not mistaken, this score means that we think bridging to Solana or Cosmos is a much higher indicator of risk than having a bridge code that has never been audited (I1011 is 2) or one that has been audited but has unresolved critical vulnerabilities (I1013 is 10). This relative weighting of risk does not feel sound to me.

Reply:
@ArjunChand13 and @vaibhavchellani   it would be great to have your thoughts on this topic.

Reply:
>The weight we give to this risk factor feels very high (I701 alone is 20). If I am not mistaken, this score means that we think bridging to Solana or Cosmos is a much higher indicator of risk than having a bridge code that has never been audited (I1011 is 2) or one that has been audited but has unresolved critical vulnerabilities (I1013 is 10). This relative weighting of risk does not feel sound to me.

I agree that the weights need to be calibrated.

Reply:
>is a much higher indicator of risk than having a bridge code that has never been audited (I1011 is 2) or one that has been audited but has unresolved critical vulnerabilities (I1013 is 10). This relative weighting of risk does not feel sound to me.

I will update that. The auditing scoring is hard to parse, and hence people could come up with the wrong impression.

Reply:
I have updated both the code audit scoring and the well known programming language scoring. Essentially now, if you haven't audited your on-chain or off-chain components, then the project will have risk score of 70 just due to code auditing. I have reduced the "not Solidity" score to 5.

Reply:
Should we include a list of languages that are "common" other than Solidity? Are there other languages that there can be enough "eyes on the code" to ensure good review?

Reply:
@ermyas @drinkcoffee I like the intent behind these risks, but agree there is some work to do on refining both how they are presented and the scores used.

Ermyas, the key point here is that software that is new, has few eyes on it, is used more than it is understood, or (as is very common on blockchains) has been built on an unstable/untested/buggy framework is inherently more risky than software that has had some sort of validation. Sometimes validation can take the form of maximising the opportunity for people to find problems (but long-standing and highly damaging bugs still crop up on the net).

I suggest keeping these two risks, but calling them something other than "well known". I'd probably just dump that term altogether and sticking with "Platform Risk Score" and "Smart Contract Programming Language Risk Score". We could then iterate the values and detailed approach over time.